virttest/utils_v2v.py

```diff
@@ -1125,3 +1199,80 @@ def v2v_mount(src, dst='v2v_mount_point'):
             (src, mount_point))
 
     return mount_point
+
+
+def create_virsh_instance(
+        hypervisor,
+        uri,
+        remote_ip,
+        remote_user,
+        remote_pwd,
+        debug=True):
+    """
+    Create a virsh instance for all hypervisors(VMWARE, XEN, KVM)
+
+    :param hypervisor: a hypervisor type
+    :param uri: uri of libvirt instance to connect to
+    :param remote_ip: Hostname/IP of remote system to ssh into (if any)
+    :param remote_user: Username to ssh in as (if any)
+    :param remote_pwd: Password to use, or None for host/pubkey
+    :param debug: Whether to enable debug
+    """
+    logging.debug("virsh connection info: uri=%s ip=%s", uri, remote_ip)
+    if hypervisor == 'kvm':
+        v2v_virsh = virsh
+    else:
+        virsh_dargs = {'uri': uri,
+                       'remote_ip': remote_ip,
+                       'remote_user': remote_user,
+                       'remote_pwd': remote_pwd,
+                       'debug': debug}
+        v2v_virsh = virsh.VirshPersistent(**virsh_dargs)
+    return v2v_virsh
+
+
+def close_virsh_instance(virsh_instance=None):
+    """
+    Close a virsh instance
+
+    :param v2v_virsh_instance: a virsh instance
+    """
+
+    if virsh_instance:
+        virsh_instance.close_session()
+
+
+def get_all_ifaces_info(vm_name, virsh_instance):
+    """
+    Get all interfaces' information
+
+    :param vm_name: vm's name
+    :param v2v_virsh_instance: a virsh instance
+    """
+    vmxml = vm_xml.VMXML.new_from_dumpxml(
+        vm_name, virsh_instance=virsh_instance)
+    interfaces = vmxml.get_iface_all()
+    if len(interfaces.keys()) == 0:
+        raise exceptions.TestError("Not found mac address for vm %s" % vm_name)
+
+    # vm_ifaces = [(mac, type), ...]
+    vm_ifaces = []
+    for mac, iface in interfaces.items():
+        vm_ifaces.append((mac, iface.get('type')))
+
+    logging.debug("Iface information for vm %s: %s", vm_name, vm_ifaces)
+    return vm_ifaces
+
+
+def v2v_supported_option(opt_str):
+    """
+    Check if an cmd option is supported by v2v
+
+    :param opt_str: option string for checking
+    """
+    cmd = 'virt-v2v --help'
+    result = process.run(cmd, verbose=True, ignore_status=True)
+    result.stdout = results_stdout_52lts(result)
```

Comment:
I'm not sure about this, but is it still necessary to use function 'results_stdout_52lts' now?

Reply:
I'm not sure, either. But since nobody talked about this, it's better to keep it.